core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/extras/array/ByteArrayCodec.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.type.codec.extras.array;
+
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import com.datastax.oss.driver.api.core.type.DataTypes;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.internal.core.type.codec.SimpleBlobCodec;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import net.jcip.annotations.Immutable;
+
+/**
+ * A codec that maps the CQL type {@code list<byte>} to the Java type {@code byte[]}.
+ *
+ * <p>Note that this codec is not suitable for reading CQL blobs as byte arrays; you should use
+ * {@link SimpleBlobCodec} for that.
```

Comment:
We have 3 different codecs dealing with bytes:

1. `BlobCodec` : `blob <-> ByteBuffer`
1. `SimpleBlobCodec` : `blob <-> byte[]`
1. `ByteArrayCodec` : `list<tinyint> <-> byte[]`